codes/codes.go

```diff
@@ -196,3 +242,24 @@ func (c *Code) UnmarshalJSON(b []byte) error {
 	}
 	return fmt.Errorf("invalid code: %q", string(b))
 }
+
+// FrameworkPermitted returns true if gRPC framework code is allowed to
+// synthesize this error code, even when the remote end did not explicitly send
+// it as a reply to the RPC.
+//
+// Note that frameworks other than core gRPC may return other error codes.
+// Consult their documentation as appropriate.
+func (c *Code) FrameworkPermitted() bool {
```

Comment:
How are you planning to use this function?

Reply:
Good point - will remove